ckanext/saml2/templates/user/edit_user_form.html

```diff
@@ -1,24 +1,30 @@
 {% import 'macros/form.html' as form %}
 
-{% if c.allow_user_changes %}
-    {% if not c.is_allow_update %}
-        {% set attrs = {'readonly': 'readonly'} %}
+{% if c.allow_user_changes is none and c.is_allow_update is none %}
+    {% set attrs = {} %}
+{% else %}
+    {% if c.allow_user_changes %}
+        {% if not c.is_allow_update %}
+            {% set attrs = {'readonly': 'readonly'} %}
+        {% else %}
+            {% set attrs = {} %}
+        {% endif %}
     {% else %}
-        {% set attrs = {} %}
+        {% set attrs = {'readonly': 'readonly'} %}
     {% endif %}
-{% else %}
-    {% set attrs = {'readonly': 'readonly'} %}
 {% endif %}
 
+
 {% block user_edit_form %}
     <form id="user-edit-form" class="dataset-form form-horizontal" method="post" action="{{ action }}">
       {{ form.errors(error_summary) }}
       {% block user_edit_form_details %}
           <fieldset>
             <legend>{{ _('Change details') }}</legend>
             {% block user_custom_data %}
-                {% if c.allow_user_changes|lower == 'true' %}
-                    {{ form.checkbox("user_custom_profile_data", id="user_custom_profile_data", label="I want to provide custom profile data", value=c.is_allow_update, classes=["user_custom_profile_data"], checked=c.is_allow_update) }}
+                {% if c.allow_user_changes is not none and c.allow_user_changes %}
```

Comment:
I think, that you need this js if `c.allow_user_changes` is set to true. Or, even better - you can have it enabled all the time!